internal/blocktx/store/postgresql/verify_merkle_roots.go

```diff
@@ -0,0 +1,76 @@
+package postgresql
+
+import (
+	"context"
+	"database/sql"
+	"encoding/hex"
+	"errors"
+
+	"github.com/bitcoin-sv/arc/internal/blocktx/store"
+	"github.com/bitcoin-sv/arc/pkg/blocktx/blocktx_api"
+)
+
+func (p *PostgreSQL) VerifyMerkleRoots(
+	_ context.Context,
+	merkleRoots []*blocktx_api.MerkleRootVerificationRequest,
+	maxAllowedBlockHeightMismatch int,
+) (*blocktx_api.MerkleRootVerificationResponse, error) {
+	qTopHeight := `
+		SELECT MAX(b.height) FROM blocks b WHERE b.orphanedyn = false
+	`
+
+	var topHeight int64
+
+	err := p.db.QueryRow(qTopHeight).Scan(&topHeight)
+	if err != nil {
+		if errors.Is(err, sql.ErrNoRows) {
+			return nil, store.ErrNotFound
+		}
+		return nil, err
+	}
+
+	qMerkleRoot := `
+		SELECT b.height FROM blocks b WHERE b.merkleroot = $1 AND b.height = $2 AND b.orphanedyn = false
+	`
+
+	var unverifiedBlockHeights []uint64
+
+	for _, mr := range merkleRoots {
+		var hash []byte
```

Comment:
It seems this hash is not used anywhere. Can we remove it?

Reply:
Done